mypy/checker.py

```diff
@@ -1890,7 +1893,6 @@ def check_compatibility_all_supers(self, lvalue: RefExpr, lvalue_type: Optional[
                         # Only show one error per variable; even if other
                         # base classes are also incompatible
                         return True
-                    break
```

Comment:
I think in order to maintain consistency with the current behavior, you still need this break after last _non-immediate_ base.

Reply:
EDIT: (non-immediate -> immediate).

Reply:
See my replies below.